streams/src/main/java/org/apache/kafka/streams/Topology.java

```diff
@@ -77,6 +77,166 @@ protected Topology(final InternalTopologyBuilder internalTopologyBuilder) {
     public enum AutoOffsetReset {
```

Comment:
We should mark this enum as `@Deprecated` plus all the methods which use it (or did you not do this on purpose yet, but want to do it a follow up PR?)

Reply:
A charitable assumption, and this was an honest oversight. I am adding this now, thank you.

Reply:
The existing methods `addSource(...)` which take this enum as parameter, should also get the `@Deprecated` annotation, and corresponding `@deprecated` JavaDocs tag (linking to the newly added methods, which take the new `AutoOffsetReset` class.